substratevm/mx.substratevm/mx_substratevm.py

```diff
@@ -1054,7 +1054,9 @@ def _native_image_launcher_extra_jvm_args():
     Gets the extra JVM args needed for running com.oracle.svm.driver.NativeImage.
     """
     # Support for running as Java module
-    res = []
+    res = [
+        f'-XX:{max_heap_size_flag}', '-XX:MaxRAMPercentage=80',
```

Comment:
If you set `-XX:MaxHeapSize=` then `MaxRAMPercentage` isn't being used, so it makes no sense to set them both. `MaxRAMPercentage` only comes into play if no values for the heap size are being given (in which case the JVM uses a percentage of physical memory). It makes sense to set it when `MaxRAM` is being set - which overrides physical memory detection -, since it then uses the set percentage for the heap size as no explicit heap size setting is been done otherwise.